docs/advanced/hmr.md

```diff
@@ -145,6 +124,29 @@ if (environment.hmr) {
 }
 ```
 
+### Update src/app/app.module.ts to manage the state in HMR lifecycle:
+
+```ts
+import { StateStream } from '@ngxs/store';
+import { NgxsHmrPlugin, NgxsStoreSnapshot } from '@ngxs/hmr-plugin';
+
+@NgModule({ .. })
+export class AppModule implements NgxsHmrPlugin {
+ 
+  public hmrNgxsStoreOnInit(state: StateStream, snapshot: NgxsStoreSnapshot) {
+    console.log('[NGXS HMR]: Current state', state.getValue());
+    console.log('[NGXS HMR]: Previous state', snapshot);
+    state.next({ ...state.getValue(), ...snapshot });
+  }
+
+  public hmrNgxsStoreOnDestroy(state: StateStream): NgxsStoreSnapshot  {
+    console.log('[NGXS HMR]: Return the necessary HMR state for reuse...');
+    return state.getValue();
+  }
+
+}
+```
```

Comment:
I would prefer if we could get the users away from direct interaction with the state stream.
I think we should rather give them a `StateContext` for these interactions:
```TS
import { StateContext } from '@ngxs/store';
import { NgxsHmrLifeCycle, NgxsStoreSnapshot } from '@ngxs/hmr-plugin';

@NgModule({ .. })
export class AppModule implements NgxsHmrLifeCycle {

  // If this hook is not supplied then it defaults to doing a patchState with the hmr snapshot
  public hmrNgxsStoreOnInit(ctx: StateContext<any>, snapshot: NgxsStoreSnapshot) {
    console.log('[NGXS HMR]: Current state', cts.getState();
    console.log('[NGXS HMR]: Previous state', snapshot);
    // Added something exra to the state here for demo purposes
    ctx.patchState({ ...snapshot, hmr: { status: 'loaded' } });
    // They could use setState if they want to replace the entire state
    // They could also potentially even dispatch some custom actions that let the State Classses choose what to do with the hmr state
  }

  // If this hook is not supplied then it defaults to snapshotting the entire state
  public hmrNgxsStoreOnDestroy(ctx: StateContext<any>): NgxsStoreSnapshot {
    console.log('[NGXS HMR]: Return the necessary HMR state for reuse...');
    return ctx.getState();
  }
}
```
I am in two minds about the ctx for the destroy because that would also make `setState`, `patchState`, and `dispatch` available. I can't think of a usecase for using those at the destroy phase. That being said, maybe it could be useful to have. Not sure what you could still do while the destroy is happening.

PS. Note that I also renamed `NgxsHmrPlugin` to `NgxsHmrLifeCycle` above.